fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,248 @@
+            interactor = interactor,
+        )
+
+        activity.themeManager.applyStatusBarTheme(activity)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        consumeFrom(onboardingStore) { state ->
+            onboardingView?.update(state.onboardingState, requireComponents.fenixOnboarding.config)
+        }
+
+        homeMenuView = HomeMenuView(
+            view = view,
+            context = view.context,
+            lifecycleOwner = viewLifecycleOwner,
+            homeActivity = activity as HomeActivity,
+            navController = findNavController(),
+            menuButton = WeakReference(binding.menuButton),
+            hideOnboardingIfNeeded = interactor::onStartBrowsingClicked,
+        ).also { it.build() }
+
+        tabCounterView = TabCounterView(
+            context = requireContext(),
+            browsingModeManager = browsingModeManager,
+            navController = findNavController(),
+            tabCounter = binding.tabButton,
+        )
+
+        toolbarView?.build()
+
+        PrivateBrowsingButtonView(
+            button = binding.privateBrowsingButton,
+            browsingModeManager = browsingModeManager,
+            onClick = { mode ->
+                interactor.onPrivateModeButtonClicked(mode, userHasBeenOnboarded = false)
+            },
+        )
+
+        consumeFrom(store) {
+            tabCounterView?.update(it)
+        }
+
+        searchSelectorBinding.set(
+            feature = SearchSelectorBinding(
+                context = view.context,
+                binding = binding,
+                browserStore = store,
+                searchSelectorMenu = searchSelectorMenu,
+            ),
+            owner = viewLifecycleOwner,
+            view = binding.root,
+        )
+
+        searchSelectorMenuBinding.set(
+            feature = SearchSelectorMenuBinding(
+                context = view.context,
+                interactor = interactor,
+                searchSelectorMenu = searchSelectorMenu,
+                browserStore = store,
+            ),
+            owner = viewLifecycleOwner,
+            view = view,
+        )
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        if (browsingModeManager.mode == BrowsingMode.Private) {
+            activity?.window?.setBackgroundDrawableResource(R.drawable.private_home_background_gradient)
+        }
+
+        hideToolbar()
+    }
+
+    override fun onPause() {
+        super.onPause()
+
+        if (browsingModeManager.mode == BrowsingMode.Private) {
+            activity?.window?.setBackgroundDrawable(
+                ColorDrawable(
+                    ContextCompat.getColor(
+                        requireContext(),
+                        R.color.fx_mobile_private_layer_color_1,
+                    ),
+                ),
+            )
+        }
```

Comment:
wondering if `HomeActivity` should just do this for us.

Reply:
Not sure, filed [Bug 1835335 - Investigate setting the background drawable color in the HomeActivity](https://bugzilla.mozilla.org/show_bug.cgi?id=1835335) as a followup.